examples/run/cpu/misaligned_truth_fitting_example.cpp

```diff
@@ -0,0 +1,219 @@
+    auto field = detray::bfield::create_const_field<traccc::scalar>(B);
+
+    // Read the detector
+    detray::io::detector_reader_config reader_cfg{};
+    reader_cfg.add_file(
+        traccc::io::get_absolute_path(detector_opts.detector_file));
+    if (!detector_opts.material_file.empty()) {
+        reader_cfg.add_file(
+            traccc::io::get_absolute_path(detector_opts.material_file));
+    }
+    if (!detector_opts.grid_file.empty()) {
+        reader_cfg.add_file(
+            traccc::io::get_absolute_path(detector_opts.grid_file));
+    }
+    const auto [host_det, names] =
+        detray::io::read_detector<host_detector_type>(host_mr, reader_cfg);
+
+    /// Create a "misaligned" context in the transform store
+    using xf_container = host_detector_type::transform_container;
+    using xf_vector = xf_container::base_type;
+
+    const xf_container& default_xfs = host_det.transform_store();
+    xf_vector misaligned_xfs;
+    misaligned_xfs.reserve(default_xfs.size());
+    for (const auto& xf : default_xfs) {
+        misaligned_xfs.push_back(xf);
+    }
+    xf_container* ptr_default_xfs = const_cast<xf_container*>(&default_xfs);
+    ptr_default_xfs->add_context(misaligned_xfs);
+
+    /*****************************
+     * Do the reconstruction
+     *****************************/
+
+    /// Standard deviations for seed track parameters
+    static constexpr std::array<scalar, e_bound_size> stddevs = {
+        0.03f * traccc::unit<scalar>::mm,
+        0.03f * traccc::unit<scalar>::mm,
+        0.017f,
+        0.017f,
+        0.001f / traccc::unit<scalar>::GeV,
+        1.f * traccc::unit<scalar>::ns};
+
+    // Fitting algorithm objects
+    // Alg0
+    traccc::fitting_config fit_cfg0(fitting_opts);
+    fit_cfg0.propagation = propagation_opts;
+    fit_cfg0.propagation.context = detray::geometry_context{0};
+    traccc::host::kalman_fitting_algorithm host_fitting0(
+        fit_cfg0, host_mr, copy, logger().clone("FittingAlg0"));
+    // Alg1
+    traccc::fitting_config fit_cfg1(fitting_opts);
+    fit_cfg1.propagation = propagation_opts;
+    fit_cfg1.propagation.context = detray::geometry_context{1};
+    traccc::host::kalman_fitting_algorithm host_fitting1(
+        fit_cfg1, host_mr, copy, logger().clone("FittingAlg1"));
+
+    // Seed generators
+    traccc::seed_generator<host_detector_type> sg0(
+        host_det, stddevs, 0, fit_cfg0.propagation.context);
+    traccc::seed_generator<host_detector_type> sg1(
+        host_det, stddevs, 0, fit_cfg1.propagation.context);
+
+    // Iterate over events
+    for (auto event = input_opts.skip;
+         event < input_opts.events + input_opts.skip; ++event) {
+
+        // Truth Track Candidates
+        traccc::event_data evt_data(input_opts.directory, event, host_mr,
+                                    input_opts.use_acts_geom_source, &host_det,
+                                    input_opts.format, false);
+
+        // For the first half of events run Alg0
```

Comment:
Doesn't it make more sense run entire events for each context so we can compare the results between aligned and misaligned?

Reply:
Probably the current design makes more sense if we resolve the alignment with the first half and apply the aligned context to the rest of tracks.